modules/parent-join/src/main/java/org/elasticsearch/join/mapper/MetaJoinFieldMapper.java

```diff
@@ -92,22 +62,17 @@ public String typeName() {
 
         @Override
         public IndexFieldData.Builder fielddataBuilder(String fullyQualifiedIndexName, Supplier<SearchLookup> searchLookup) {
-            failIfNoDocValues();
-            return new SortedSetOrdinalsIndexFieldData.Builder(name(), CoreValuesSourceType.BYTES);
+            throw new UnsupportedOperationException("Cannot load field data for metadata field [" + NAME + "]");
         }
 
         @Override
         public ValueFetcher valueFetcher(MapperService mapperService, SearchLookup searchLookup, String format) {
-            throw new UnsupportedOperationException("Cannot fetch values for metadata field [" + typeName() + "].");
+            throw new UnsupportedOperationException("Cannot fetch values for metadata field [" + NAME + "].");
         }
 
         @Override
         public Object valueForDisplay(Object value) {
-            if (value == null) {
-                return null;
-            }
-            BytesRef binaryValue = (BytesRef) value;
-            return binaryValue.utf8ToString();
+            throw new UnsupportedOperationException();
```

Comment:
I am missing why this needs to change

Reply:
It should never be called, as the mapper is just a placeholder for holding the Joiner.